schema.graphql

```diff
@@ -107,6 +121,162 @@ type UnprocessedEntity @entity {
   block: Block
 }
 
+type Service @entity {
+  id: ID!
+  name: String!
+  computeUnitsPerRelay: BigInt!
+  owner: Account!
+  addServiceMsgs: [AddServiceMsg] @derivedFrom(field: "service")
+}
+
+type AppStakeMsg @entity {
+  id: ID!
+  stake: Coin!
+  transaction: Transaction!
+  block: Block!
+  application: Application! @index
+}
+
+type AppStakeMsgService @entity {
```

Comment:
Just for clarity, these join tables are exempt from the feedback about primitive relations.